gcs/constraints/circuit/circuit_scc.cc

```diff
@@ -0,0 +1,1250 @@
+#include <gcs/constraints/circuit/circuit.hh>
+#include <gcs/innards/propagators.hh>
+#include <iostream>
+#include <list>
+#include <random>
+#include <set>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+using std::cmp_equal;
+using std::cmp_less;
+using std::cmp_less_equal;
+using std::cmp_not_equal;
+using std::cout;
+using std::get;
+using std::holds_alternative;
+using std::list;
+using std::make_optional;
+using std::make_pair;
+using std::map;
+using std::min;
+using std::move;
+using std::mt19937;
+using std::nullopt;
+using std::optional;
+using std::pair;
+using std::random_device;
+using std::set;
+using std::string;
+using std::stringstream;
+using std::to_string;
+using std::tuple;
+using std::uniform_int_distribution;
+using std::unique_ptr;
+using std::variant;
+using std::vector;
+
+using namespace gcs;
+using namespace gcs::innards;
+
+namespace
+{
+
+    struct OrderingAssumption
+    {
+        ProofFlag assumption_flag;
+        long first;
+        long middle;
+        long last;
+    };
+
+    auto select_root(long n) -> long
```

Comment:
This gives an unused variable compiler warning. Either tag it as `[[unused]]` or don't name it.